### ISSUE TITLE
CosineAnnealingLR has unexpected behavior with large step

### ISSUE THREAD

**Issue by 8enmann** [pytorch/pytorch#20405]
## 🐛 Bug

CosineAnnealingLR has unexpected behavior with large step. We use token count as the step for language modeling and this is often in the 1e9 range.

## To Reproduce
1. Construct torch.optim.lr_scheduler.CosineAnnealingLR with max_step=1e9
2. scheduler.step(x) for large x

Repro in this collab notebook
https://colab.research.google.com/drive/11htMBtisAHq7fQViW0inUMPAiA2J93t3

## Expected behavior

Behaves nicely with large numbers for step size

## Environment

 - PyTorch Version (e.g., 1.0): 1.1
 - OS (e.g., Linux): colab
 - Python version: 3.x
 - GPU models and configuration: None
